### ISSUE TITLE
Handling ENOTFOUND

### ISSUE THREAD

**Issue by atomheartother** [lmammino/tall#13]
Hello,

I know this was supposedly handled in #12 but I have`2.2.0` installed (the one with this PR) and it seems to still fail when fed a shortened URL pointing to an invalid URL. I'd like to know if it's me handling this poorly somehow or if the bug is still present.

How to reproduce:
- Get on twitter or bit.ly, and ask to shorten to an invalid URL like `z.chat`
- Feed the shortened url to `tall`: `tall("https://t.co/FqLTQ7TRYS")`
- The promise doesn't reject anything, instead the node process stops with error:
```
Error: getaddrinfo ENOTFOUND z.chat z.chat:80
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:58:26)
Emitted 'error' event at:
    at Socket.socketErrorListener (_http_client.js:397:9)
    at Socket.emit (events.js:193:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:59:3)
    at processTicksAndRejections (internal/process/task_queues.js:81:17)
```

Am I missing something, am I handling the promise rejection wrong?
```js
  const promises = urls.map((shortUrl, idx) =>
    tall(shortUrl).then(longUrl => ({
      shortUrl,
      longUrl,
      idx
    }))
  );

  Promise.all(promises)
    .then(results => {
        // Handle success
    })
    .catch(e => {
      // THIS NEVER GETS CALLED
      log("The elusive buggerino!");
      log(`Total message: ${text}`);
      log(`URL we tried shortening: ${shortUrl}`);
      log(e);
    });
```

Thanks for your time

**Comment by FiveBoroughs** [comment]
I'm having the same issue, @lmammino this is blocking if any wrong URL can crash your program, any help?

**Comment by lmammino** [comment]
Closed by #15  - Thanks @FiveBoroughs  for your PR :)
